src/mkdocstrings/handlers/__init__.py

```diff
@@ -66,10 +67,44 @@ def do_highlight(
     result = highlighter.highlight(src=src, language=language, linestart=line_start, inline=inline)
 
     if inline:
-        return do_mark_safe(result.text)
+        return f'<code class="highlight language-{language}">{do_mark_safe(result.text)}</code>'
```

Comment:
All of this needs to be `do_mark_safe`, otherwise the output is `&lt;code class="highlight` etc.